lib/Onyx.js

```diff
@@ -997,7 +998,11 @@ function merge(key, value) {
                 // don't apply these changes again
                 delete mergeQueue[key];
 
-                return set(key, modifiedData);
+                // We don't wait until the merge is written to storage.
+                // Instead, we immediately notify subscribers and then return the merge promise
+                const mergePromise = OnyxMerge.merge(key, changes, modifiedData);
+
+                return notify(key, modifiedData, true).then(() => mergePromise);
```

Comment:
Similar to the comment in `set` - we don't need `notify` to be a promise returning method 

We can just go for:

```js
// We relay changes to subscribers optimistically to avoid lag (think UI updates)
notify(key, modifiedData, 'merge');
return OnyxMerge.merge(key, changes, modifiedData);
```

We don't need a comment explaining "what" and "how" but "why"